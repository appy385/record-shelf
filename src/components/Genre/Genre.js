import React from 'react';
import PropTypes from 'prop-types';
import './Genre.scss';
import Song from '../Song/Song';
import Bollywood from '../../assets/genre-bollywood.png';
import Country from '../../assets/genre-country.png';
import Pop from '../../assets/genre-pop.png';
import Rock from '../../assets/genre-rock.png';

const Genre = ({ genreSongs, handleLike }) => {
  const renderSwitch = (param) => {
    switch (param) {
      case 'Bollywood': return (
        <div className="genre-image-container">
          <img className="genre-image" src={Bollywood} alt="song" />
          <div className="genre-image-name">{`${param}`}</div>
        </div>
      );
      case 'Country': return (
        <div className="genre-image-container">
          <img className="genre-image" src={Country} alt="song" />
          <div className="genre-image-name">{`${param}`}</div>
        </div>
      );
      case 'Pop': return (
        <div className="genre-image-container">
          <img className="genre-image" src={Pop} alt="song" />
          <div className="genre-image-name">{`${param}`}</div>
        </div>
      );
      case 'Rock': return (
        <div className="genre-image-container">
          <img className="genre-image" src={Rock} alt="song" />
          <div className="genre-image-name">{`${param}`}</div>
        </div>
      );
      default: return (
        <div className="genre-image-container">
          <img className="genre-image" src={Country} alt="song" />
          <div className="genre-image-name">{`${param}`}</div>
        </div>
      );
    }
  };

  return (
    <>
      {Object.keys(genreSongs).map((genre) => (
        <React.Fragment key={genre}>
          <div className="genre-category">
            {renderSwitch(genre)}
            <div className="genre-row" data-testid="genre-row">
              {genreSongs[genre].map((song) => (
                <Song key={song.id} song={song} handleLike={handleLike} />))}
            </div>
          </div>
        </React.Fragment>
      ))}
    </>
  );
};

const songShape = PropTypes.shape({
  id: PropTypes.string.isRequired,
  albumArtUrl: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  artist: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  genre: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  publishedAt: PropTypes.string.isRequired,
});

Genre.propTypes = {
  genreSongs: PropTypes.objectOf(PropTypes.arrayOf(songShape)).isRequired,
  handleLike: PropTypes.func.isRequired,

};

export default Genre;
