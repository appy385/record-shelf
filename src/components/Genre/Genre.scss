.genre-category {
    display: flex;
    flex-direction: column;
    margin-bottom: 5rem;
}
.genre-image-container {
    display: flex;
}
.genre-image-name {
    padding: 1rem 1.5rem 1rem 1.5rem;
    border: solid 1px black;
    background-color: #4C63AA;
    box-shadow: 0 0.25rem 0.25rem 0 #9a9a9a;
    align-self: center;
    border-radius: 1.5rem;

}
.genre-image {
    border-radius: 50%;
    margin-top: 1rem;
    margin-bottom: 1rem;
    width: calc((100% - (8 * 0.5rem + 10*1rem)) / 5);
}
.genre-row {
    display: flex;
    flex-wrap: wrap;
}

@media (max-width: 30rem) {
    .genre-image{
        width: 40%;
    }
}