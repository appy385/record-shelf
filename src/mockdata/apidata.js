export const mockgetSongsApi = {
  data: {
    data: [
      {
        albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6',
        artist: { id: 'b22d7ed5-3ded-4958-a036-60453b7fc102', name: 'Neha Kakkar' },
        genre: { id: '37146752-2c95-4204-a845-a32f3cdce9c8', name: 'Bollywood' },
        id: '07075cdb-9e5c-41ec-8c09-9a1322dd92d8',
        name: 'Aankh Marey (From Simmba)',
        publishedAt: '2018-12-06T00:00:00',
      },
      {
        albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273d9009cc16eddbc6664508cd8',
        artist: { id: 'a9d57d30-182f-4297-b5d9-f1e765c86935', name: 'Phillip Phillips' },
        genre: { id: '8f63f8b8-a803-45da-8b2d-26a32ede9230', name: 'Country' },
        id: '08d7e81e-1c34-4a60-a70c-2ec11c8a6687',
        name: 'Gone, Gone, Gone',
        publishedAt: '2013-01-01T00:00:00',
      },

    ],
  },

};

export const newData = {};
