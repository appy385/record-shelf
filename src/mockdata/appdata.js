export const mockSongData = {
  albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273d9009cc16eddbc6664508cd8',
  artist: { id: 'a9d57d30-182f-4297-b5d9-f1e765c86935', name: 'Phillip Phillips' },
  genre: { id: '8f63f8b8-a803-45da-8b2d-26a32ede9230', name: 'Country' },
  id: '08d7e81e-1c34-4a60-a70c-2ec11c8a6687',
  name: 'Gone, Gone, Gone',
  publishedAt: '2013-01-01T00:00:00',
};

export const newS = {};
