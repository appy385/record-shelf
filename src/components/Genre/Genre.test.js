import React from 'react';
import {
  render, screen,
} from '@testing-library/react';
import Genre from './Genre';
import { mockSongsByCategory } from '../../mockdata/appdata';

describe(Genre.name, () => {
  test('should match snapshot', async () => {
    const mockHandleLike = jest.fn();
    const { container } = render(<Genre
      genreSongs={mockSongsByCategory}
      handleLike={mockHandleLike}
    />);
    expect(container).toMatchSnapshot();
  });

  test('should have number of genres equal to mock data genre', async () => {
    const mockHandleLike = jest.fn();
    render(<Genre genreSongs={mockSongsByCategory} handleLike={mockHandleLike} />);
    const element = screen.getAllByTestId('genre-row');
    const { length } = Object.keys(mockSongsByCategory);
    expect(element.length).toBe(length);
  });
});
