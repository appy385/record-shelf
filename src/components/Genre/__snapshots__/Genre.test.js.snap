// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre should match snapshot 1`] = `
<div>
  <div
    class="genre-category"
  >
    <div
      class="genre-image-container"
    >
      <img
        alt="song"
        class="genre-image"
        src="genre-country.png"
      />
      <div
        class="genre-image-name"
      >
        Country
      </div>
    </div>
    <div
      class="genre-row"
      data-testid="genre-row"
    >
      <div
        class="song-card"
        data-testid="song-card"
      >
        <img
          alt="song"
          class="song-image"
          src="https://i.scdn.co/image/ab67616d0000b273d9009cc16eddbc6664508cd8"
        />
        <div
          class="song-container"
        >
          <div>
            <div
              class="song-name"
            >
              Gone, Gone, Gone
            </div>
            <div
              class="song-artist"
            >
              Phillip Phillips
            </div>
          </div>
          <div
            class="song-image-container"
          >
            <input
              alt="text"
              class="song-heart-button"
              data-testid="gray-heart"
              src="heart-gray.svg"
              type="image"
            />
            <div
              class="song-likes"
            >
              <span
                class="song-like-count"
              >
                3
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="genre-category"
  >
    <div
      class="genre-image-container"
    >
      <img
        alt="song"
        class="genre-image"
        src="genre-country.png"
      />
      <div
        class="genre-image-name"
      >
        Bollywoord
      </div>
    </div>
    <div
      class="genre-row"
      data-testid="genre-row"
    >
      <div
        class="song-card"
        data-testid="song-card"
      >
        <img
          alt="song"
          class="song-image"
          src="https://i.scdn.co/image/ab67616d0000b273b9c46d2b8d5604c33cc578b7"
        />
        <div
          class="song-container"
        >
          <div>
            <div
              class="song-name"
            >
              Illegal Weapon 2.0 (From Street Dancer 3D)
            </div>
            <div
              class="song-artist"
            >
              Jasmine Sandlas
            </div>
          </div>
          <div
            class="song-image-container"
          >
            <input
              alt="text"
              class="song-heart-button"
              data-testid="gray-heart"
              src="heart-gray.svg"
              type="image"
            />
            <div
              class="song-likes"
            >
              <span
                class="song-like-count"
              >
                3
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
